File: 2022/d1/d1p1.py
running_total = 0
current_max = 0

with open("d1.in") as file:
    for line in file.readlines():
        if line == "\n":
            if running_total > current_max:
                current_max = running_total
            running_total = 0
            print(current_max)
        else:
            running_total += int(line)

print(current_max)

File: 2022/d1p1.cpp
#include <iostream>
#include <string>

int main(void) {
    int running_total = 0;
    int current_max = 0;
    std::string line;
    while (std::getline(std::cin, line)) {
        if (line.empty()) {
            if (running_total > current_max) {
                current_max = running_total;
            }
            running_total = 0;
        } else {
            running_total += std::stoi(line);
        }
    }
    std::cout << current_max << std::endl;
    return 0;
}
